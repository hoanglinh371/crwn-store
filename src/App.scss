* {
  box-sizing: border-box;
}

a {
  color: #000;
  text-decoration: none;
}

ul li {
  list-style: none;
}

html {
  font-size: 100%;
}

body {
  padding: 20px 40px;
}

@media screen and (max-width: 600px) {
  html {
    font-size: 75%;
  }

  body {
    padding: 10px;
  }
}
