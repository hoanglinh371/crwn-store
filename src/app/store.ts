import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
