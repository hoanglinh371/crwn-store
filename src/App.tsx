import React from 'react';
import { Route, Routes } from 'react-router-dom';

import './App.scss';
import { publicRoutes } from './routes';

import Navigation from './components/Navigation';

const App: React.FC = () => {
  return (
    <React.Fragment>
      <Navigation />
      <Routes>
        {publicRoutes.map((route, i) => {
          const { path, element } = route;
          return <Route key={i} path={path} element={element} />;
        })}
      </Routes>
    </React.Fragment>
  );
};

export default App;
