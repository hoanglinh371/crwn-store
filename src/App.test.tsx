import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from 'src/app/store';
import App from './App';

describe('App component', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );

  it('Should_MatchSnapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
