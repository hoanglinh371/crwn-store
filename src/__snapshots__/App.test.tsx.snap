// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="overlay"
  >
    <div
      class="container"
    />
  </div>
</div>
`;
